src/components/course/data/tests/utils.test.jsx

```diff
@@ -970,3 +970,188 @@ describe('getMissingApplicableSubsidyReason', () => {
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('returns the minimal course metadata with typical values, happy path', () => {
+    const minimaCourseMetadata = transformedCourseMetadata({
+      transformed,
+      coursePrice,
+      courseRunKey: mockCourseRunKey,
+      currency: mockCurrency,
+    });
+    expect(minimaCourseMetadata).toEqual(expectedValue);
+  });
+  it('fallsback to activeCourseRun if no course run key matches', () => {
+    const minimaCourseMetadata = transformedCourseMetadata({
+      transformed,
+      coursePrice,
+      courseRunKey: mockActiveCourseRunKey,
+      currency: mockCurrency,
+    });
+    const updatedExpectedValue = {
+      ...expectedValue,
+      startDate: mockActiveCourseRunStartDate,
+      duration: '16 Weeks',
+    };
+    expect(minimaCourseMetadata).toEqual(updatedExpectedValue);
+  });
+  it.each([
+    {
+      organizationShortCodeOverride: null,
+      organizationLogoOverrideUrl: null,
+      owners: [{
+        name: mockOrgName,
+        marketingUrl: mockOrgMarketingUrl,
+        logoImageUrl: mockLogoImageUrl,
+        uuid: 'test-uuid',
+        key: 'test-key',
+      }],
+      organization: {
+        logoImgUrl: 'https://fake-logo.url',
+        marketingUrl: 'https://fake-mktg.url',
+        name: 'Fake Org Name',
+      },
+    },
+    {
+      organizationShortCodeOverride: 'test-short-code-override',
+      organizationLogoOverrideUrl: 'test-logo-override',
+      owners: [{
+        name: mockOrgName,
+        marketingUrl: mockOrgMarketingUrl,
+        logoImageUrl: mockLogoImageUrl,
+        uuid: 'test-uuid',
+        key: 'test-key',
+      }],
+      organization: {
+        logoImgUrl: 'test-logo-override',
+        marketingUrl: 'https://fake-mktg.url',
+        name: 'test-short-code-override',
+      },
+    },
+  ])('handles organizations correctly when values are %s', ({
+    organizationShortCodeOverride,
+    organizationLogoOverrideUrl,
+    owners,
+    organization,
+  }) => {
+    const updatedTransformedData = {
+      ...transformed,
+      organizationLogoOverrideUrl,
+      organizationShortCodeOverride,
+      owners,
+    };
+    const minimaCourseMetadata = transformedCourseMetadata({
+      transformed: updatedTransformedData,
+      coursePrice,
+      courseRunKey: mockCourseRunKey,
+      currency: mockCurrency,
+    });
+    expect(minimaCourseMetadata.organization).toEqual(organization);
+  });
+  it.each([{
+    courseRuns: [{
+      key: mockCourseRunKey,
+      weeksToComplete: mockWeeksToComplete,
+      start: mockCourseRunStartDate,
+    }],
+    duration: '8 Weeks',
+  },
+  {
+    courseRuns: [{
+      key: mockCourseRunKey,
+      weeksToComplete: 1,
+      start: mockCourseRunStartDate,
+    }],
+    duration: '1 Week',
+  },
+  {
+    courseRuns: [],
+    duration: '-',
+  },
+  ])('handles duration correctly when values are %s', ({
+    courseRuns,
+    duration,
+  }) => {
+    let updatedTransformed = {
```

Comment:
nit: could keep this as a `const` since it's an object;

```jsx
const updatedTransformed = { ...transformed, courseRuns };
if (courseRuns.length === 0) {
  updatedTransformed.activeCourseRun = null;
}
```